### PR TITLE
Navigations can now have their animations controlled with a parameter

### PR DESCRIPTION
**Pull request by Velin92**
In this PR, a refactor has been done on the navigation coordinators, so that they can now have all their navigation functions be fine tuned to show an animation or not.

This required also to turn the compact layout into a computed binding.

### CODE REVIEW

File: ElementX/Sources/Application/Navigation/NavigationCoordinators.swift [hunk 1]
```diff
@@ -23,7 +23,9 @@ class NavigationSplitCoordinator: CoordinatorProtocol, ObservableObject, CustomS
     fileprivate let placeholderModule: NavigationModule
     
     private var cancellables = Set<AnyCancellable>()
-    
+
+    var sidebarWillChangeCancellable: AnyCancellable?
```
Comment (pixlwave) [inline ElementX/Sources/Application/Navigation/NavigationCoordinators.swift:27]: Optional: maybe slightly clearer to call this
```suggestion
    var sidebarStackModulesCancellable: AnyCancellable?
```

File: ElementX/Sources/Application/Navigation/NavigationCoordinators.swift [hunk 2]
```diff
@@ -34,16 +36,21 @@ class NavigationSplitCoordinator: CoordinatorProtocol, ObservableObject, CustomS
             if let sidebarModule {
                 logPresentationChange("Set sidebar", sidebarModule)
                 sidebarModule.coordinator?.start()
+                if let observableCoordinator = sidebarModule.coordinator as? NavigationStackCoordinator {
+                    sidebarWillChangeCancellable = observableCoordinator.$stackModules.sink { [weak self] _ in
```
Comment (pixlwave) [inline ElementX/Sources/Application/Navigation/NavigationCoordinators.swift:40]: Looking at this with fresh eyes, I wonder if would be sensible to set these new cancellables to `nil`  in the `oldValue` block? 

E.g. if we replace the sidebar or detail module with something that isn't a navigation stack, then it won't get cancelled (probably wouldn't cause any harm leaving it, but seems worthwhile nonetheless).

File: ElementX/Sources/Application/Navigation/NavigationCoordinators.swift [hunk 3]
```diff
@@ -34,16 +36,21 @@ class NavigationSplitCoordinator: CoordinatorProtocol, ObservableObject, CustomS
...
-            updateCompactLayoutComponents()
         }
     }
     
     /// The currently displayed sidebar coordinator
     var sidebarCoordinator: (any CoordinatorProtocol)? {
         sidebarModule?.coordinator
     }
+
+    var detailWillChangeCancellable: AnyCancellable?
```
Comment (pixlwave) [inline ElementX/Sources/Application/Navigation/NavigationCoordinators.swift:53]:
```suggestion
    var detailStackModulesCancellable: AnyCancellable?
```

File: ElementX/Sources/Application/Navigation/NavigationCoordinators.swift [hunk 4]
```diff
@@ -141,15 +204,20 @@ class NavigationSplitCoordinator: CoordinatorProtocol, ObservableObject, CustomS
...
+            sidebarModule = NavigationModule(coordinator, dismissalCallback: dismissalCallback)
+        }
     }
     
     /// Set the coordinator to be used on the split's right pannel
     /// - Parameters:
     ///   - coordinator: the detail coordinator
     ///   - dismissalCallback: called when this particular detail coordinator has removed/replaced
-    func setDetailCoordinator(_ coordinator: (any CoordinatorProtocol)?, dismissalCallback: (() -> Void)? = nil) {
+    func setDetailCoordinator(_ coordinator: (any CoordinatorProtocol)?, animated: Bool = true, dismissalCallback: (() -> Void)? = nil) {
```
Comment (pixlwave) [inline ElementX/Sources/Application/Navigation/NavigationCoordinators.swift:220]: This one is missing the doc comment.

Comment (pixlwave) [inline ElementX/Sources/Application/Navigation/NavigationCoordinators.swift:220]: This one is missing the doc comment.

File: ElementX/Sources/Application/Navigation/NavigationCoordinators.swift [hunk 5]
```diff
@@ -328,28 +352,6 @@ class NavigationSplitCoordinator: CoordinatorProtocol, ObservableObject, CustomS
...
-            guard navigationStackCoordinator.stackModules != stackModules else { return }
-            DispatchQueue.main.async { self?.updateCompactLayoutComponents() }
-        }
-        .store(in: &cancellables)
-        
-        navigationStackCoordinator.$sheetModule.sink { [weak self] sheetModule in
-            guard navigationStackCoordinator.sheetModule != sheetModule else { return }
-            DispatchQueue.main.async { self?.updateCompactLayoutComponents() }
-        }
-        .store(in: &cancellables)
```
Comment (pixlwave) [inline ElementX/Sources/Application/Navigation/NavigationCoordinators.swift:351]: Ok looking at this it also observed the `root` and `sheet` modules. From the comments it seems sheets was pointless as we have to present them on the split coordinator (although in the future perhaps we could fix that with another `Binding`) but I think we probably also need to include the root in our observation:
```
Publishers.CombineLatest2(coordinator.$rootModule, 
                          coordinator.$stackModules)
    .sink {
    …
```
Note: I don't think the unit tests will capture this as they use the `get` which will recompute and be correct whereas SwiftUI will only recompute after receiving a `willChange`.

**Review by pixlwave — APPROVED** [review]
LGTM, but I think we need to handle the `rootModule` too.

File: ElementX/Sources/Application/Navigation/NavigationCoordinators.swift (hunk 5 above)
Comment (Velin92) [reply ElementX/Sources/Application/Navigation/NavigationCoordinators.swift:351]: Why using combine latest? We could just set the observation as we did for the stack and the sidebar modules?

**Pull request merged**